server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -124,9 +128,10 @@ public static enum OpenSearchRequestType {
      * Instantiate a new ExtensionsManager object to handle requests and responses from extensions. This is called during Node bootstrap.
      *
      * @param extensionsPath  Path to a directory containing extensions.
+     * @param additionalSettings  Additional settings to read in from extensions.yml
      * @throws IOException  If the extensions discovery file is not properly retrieved.
      */
-    public ExtensionsManager(Path extensionsPath) throws IOException {
+    public ExtensionsManager(Path extensionsPath, Set<Setting<?>> additionalSettings) throws IOException {
```

Comment:
Should we call them `defaultSettings` to emphasize those will be applied if not set?

Reply:
The default settings (w/o security installed) can be found here: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/extensions/ExtensionsManager.java#L601-L608

Its currently not using Settings in the ExtensionsManager, but it may make sense to refactor all of those into `ExtensionScopedSettings` (@saratvemulapalli for comment). I was thinking additionalSettings because that's what they are called in the `SettingsModule` for plugin defined settings if a plugin overrides `getSettings()` for adding plugin settings to `opensearch.yml`

Reply:
@cwperks instead of passing `additionalSettings` in the constructor here. I'm thinking can we get those settings in `discover()` using a getter? Correct me if I'm wrong.

Reply:
Get them from where though? `discover()` is called in the constructor of `ExtensionsManager` so I put them as a constructor arg to make sure they are available at the time `discover()` is called.

Reply:
With the current design this is fine. I'm thinking we should eventually pull out the initialization completely from ExtensionsManager. This is something which can be discussed later.